src/main/java/com/ericsson/ei/jmespath/MatchFunction.java

```diff
@@ -0,0 +1,57 @@
+package com.ericsson.ei.jmespath;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.burt.jmespath.Adapter;
+import io.burt.jmespath.JmesPathType;
+import io.burt.jmespath.function.ArgumentConstraints;
+import io.burt.jmespath.function.BaseFunction;
+import io.burt.jmespath.function.FunctionArgument;
+
+public class MatchFunction extends BaseFunction {
+
+    public MatchFunction() {
+        super(ArgumentConstraints.listOf(ArgumentConstraints.typeOf(JmesPathType.STRING),
+                ArgumentConstraints.typeOf(JmesPathType.STRING)));
+    }
+
+    /**
+     * This method match a regex pattern with a given string. It is used for
+     * extracting groupID component of GAV from purl
```

Comment:
Please specify that extracting groupId is an example usage. As it is now it sounds like it is only used for that.